pkg/upstream/cluster/loadbalancer.go

```diff
@@ -529,3 +530,39 @@ func (lb *reqRoundRobinLoadBalancer) IsExistsHosts(metadata api.MetadataMatchCri
 func (lb *reqRoundRobinLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int {
 	return lb.hosts.Size()
 }
+
+// leastActiveConnectiontLoadBalancer choose the host with the least active connection
+type leastActiveConnectionLoadBalancer struct {
+	*EdfLoadBalancer
+}
+
+func newleastActiveConnectionLoadBalancer(info types.ClusterInfo, hosts types.HostSet) types.LoadBalancer {
+	lb := &leastActiveConnectionLoadBalancer{}
+	lb.EdfLoadBalancer = newEdfLoadBalancerLoadBalancer(hosts, lb.unweightChooseHost, lb.hostWeight)
+	return lb
+}
+
+func (lb *leastActiveConnectionLoadBalancer) hostWeight(item WeightItem) float64 {
+	host := item.(types.Host)
+	return float64(host.Weight()) / float64(host.HostStats().UpstreamConnectionActive.Count()+1)
+}
+
+func (lb *leastActiveConnectionLoadBalancer) unweightChooseHost(context types.LoadBalancerContext) types.Host {
+	allHosts := lb.hosts
+	total := allHosts.Size()
+	lb.mutex.Lock()
+	defer lb.mutex.Unlock()
+	var candicate types.Host
+	randIdx := lb.rand.Intn(total)
```

Comment:
Why not refer to leastActiveRequestLoadBalancer's unweightChooseHost? For traversing an entire array, a random start position doesn't affect the result, right?

Reply:
> Why not refer to leastActiveRequestLoadBalancer's unweightChooseHost? For traversing an entire array, a random start position doesn't affect the result, right?

The only difference is leastActiveRequest use UpstreamRequestActive to choose host, leastActiveConnection use UpstreamConnectionActive.